### PR TITLE
distrho-ports: fix rpath, simplify dependencies

### PR DESCRIPTION
**Pull request by tomcur**
###### Motivation for this change
#118928 was merged while #116608 was open. This new PR introduces the rpath fix addressed in  #116608. It also removes some dependencies introduced in #118928 that seem unnecessary.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tomcur** [comment]
Pinging the main participants of these PRs: @magnetophon @kanashimia @SuperSandro2000

File: pkgs/applications/audio/distrho/default.nix [hunk 1]
```diff
@@ -34,31 +31,34 @@ stdenv.mkDerivation rec {
...
-    libgcc
-    libglvnd
-    libsecret
   ];
 
+  postFixup = ''
+    for file in \
+      "$out/lib/lv2/vitalium.lv2/vitalium.so" \
+      "$out/lib/vst/vitalium.so" \
+      "$out/lib/vst3/vitalium.vst3/Contents/x86_64-linux/vitalium.so"
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/distrho/default.nix:48]:
```suggestion
      $out/lib/lv2/vitalium.lv2/vitalium.so \
      $out/lib/vst/vitalium.so \
      $out/lib/vst3/vitalium.vst3/Contents/x86_64-linux/vitalium.so
```

Comment (tomcur) [reply pkgs/applications/audio/distrho/default.nix]: Done, thank you.

File: pkgs/applications/audio/distrho/default.nix [hunk 2]
```diff
@@ -34,31 +31,34 @@ stdenv.mkDerivation rec {
...
+  '';
+
   meta = with lib; {
     homepage = "http://distrho.sourceforge.net/ports";
     description = "Linux audio plugins and LV2 ports";
     longDescription = ''
       Includes:
         arctican-function
         arctican-pilgrim
+        CHOW
```
Comment (kanashimia) [inline pkgs/applications/audio/distrho/default.nix:61]:
```suggestion
```
https://github.com/DISTRHO/DISTRHO-Ports/commit/cb80ec030896b08471aaa0f2a43fde4b505dbef3

Comment (tomcur) [reply pkgs/applications/audio/distrho/default.nix]: I had missed this, thank you.

**Review by kanashimia — APPROVED** [review]
Plugins seem to work.

**Pull request merged**